runtime/sema/check_composite_declaration.go

```diff
@@ -86,7 +86,11 @@ func (checker *Checker) visitCompositeDeclaration(declaration *ast.CompositeDecl
 
 		for _, field := range declaration.Members.Fields() {
 			fieldName := field.Identifier.Identifier
-			member := compositeType.Members[fieldName]
+			member, ok := compositeType.Members.Get(fieldName)
+			if !ok {
+				continue
```

Comment:
Maybe catch this case with 

```suggestion
				panic(errors.NewUnreachableError())
```

Reply:
I think we need to continue in this case, since some declarations can have fields that are not part of the type. (in negative scenarios)

eg:
```rust
enum Foo: Int {
    let a: Int
}
```
This fails with a semantic error in the downstream-code. But at this point, the filed `a` is part of the declaration, but not part of the type. Hence panicking here can crash the compiler.

Reply:
Oh, interesting, good catch!